schema.py

```diff
@@ -4,14 +4,24 @@
 
 import re
 
+# The setup.py (which installs dependencies) needs access to the __version__
+# and __all__ attributes. Therefore, external dependencies and their global
+# usage need to be wrapped in a try/except.
+try:
+    from contextlib2 import ExitStack
+    exitstack = ExitStack()
```

Comment:
I don't think the `exitstack` should be global. Best to assign it in `validate`.